src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/GearListener.java

```diff
@@ -1,68 +1,57 @@
 package io.github.thebusybiscuit.slimefun4.implementation.listeners;
 
-import me.mrCookieSlime.Slimefun.SlimefunPlugin;
+import io.github.thebusybiscuit.slimefun4.implementation.tasks.JetBootsTask;
+import io.github.thebusybiscuit.slimefun4.implementation.tasks.JetpackTask;
+import io.github.thebusybiscuit.slimefun4.implementation.tasks.MagnetTask;
+import io.github.thebusybiscuit.slimefun4.implementation.tasks.ParachuteTask;
 import me.mrCookieSlime.Slimefun.Lists.SlimefunItems;
 import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.JetBoots;
 import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.Jetpack;
 import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
 import me.mrCookieSlime.Slimefun.Setup.SlimefunManager;
+import me.mrCookieSlime.Slimefun.SlimefunPlugin;
 import me.mrCookieSlime.Slimefun.api.Slimefun;
-
-import org.bukkit.Bukkit;
 import org.bukkit.entity.Player;
 import org.bukkit.event.EventHandler;
 import org.bukkit.event.Listener;
 import org.bukkit.event.player.PlayerToggleSneakEvent;
 
-import io.github.thebusybiscuit.slimefun4.implementation.tasks.JetBootsTask;
-import io.github.thebusybiscuit.slimefun4.implementation.tasks.JetpackTask;
-import io.github.thebusybiscuit.slimefun4.implementation.tasks.MagnetTask;
-import io.github.thebusybiscuit.slimefun4.implementation.tasks.ParachuteTask;
-
 public class GearListener implements Listener {
 
-	public GearListener(SlimefunPlugin plugin) {
-		plugin.getServer().getPluginManager().registerEvents(this, plugin);
-	}
+    public GearListener(SlimefunPlugin plugin) {
+        plugin.getServer().getPluginManager().registerEvents(this, plugin);
+    }
+
+    @EventHandler
+    public void onToggleSneak(PlayerToggleSneakEvent e) {
+        if (!e.isSneaking()) return;
+		Player p = e.getPlayer();
+
+        if (p.getInventory().getChestplate() != null) {
+            SlimefunItem chestplate = SlimefunItem.getByItem(p.getInventory().getChestplate());
+            if (chestplate == null || !Slimefun.hasUnlocked(p, chestplate, true)) return;
```

Comment:
So if the Player wears no chestplate you return.
The jetboots will never even get checked if you return here prematurely.